OneCommunityGlobal/HighestGoodNetworkApp
# Shuhua - create different view for !summary

# Description
<img width="681" alt="task-description" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/83729360/f06536fa-e7c4-49ea-aaa9-9f776f56e998">

## Related PRS (if any):
N/A

## Main changes explained:
- Update the file **`src/components/SummaryBar/SummaryBar.jsx`** to incorporate a different view for users exempt from summary submission


## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. Verify that Mentor users see the desired view, and summary submission function still works
  4.1. Log in as a Mentor user, and confirm that the !summary view meets the expectations
  4.2. Click the text “You don’t need to …” to submit a summary
5. Verify that the view updates properly as summary requirements change
  5.1. Log in as an Admin user, and verify that the Summary Bar shows the default view
  5.2. Change the Weekly Summary requirement: Go to View Profile → Volunteering Times tab, set "**Weekly Summary Options**" to "**Not Required**", and save the change
  5.3. Go back to the Dashboard and verify the updated view
  5.4. (Optional) Toggle the "**Weekly Summary Options**" back and see the updated view
6. (Optional) Verify step 5 with a user other than Mentor/Admin

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/83729360/f3ee7431-45c2-4892-bc1a-f90ca15803da



## Note:
1. Use a backslash to escape `!` if you switch branches in the terminal: 
  `git checkout -b Shuhua-create-different-view-\!summary`
2. These 400 bad request errors with the message “MongoError: QueryExceededMemoryLimitNoDiskUseAllowed” exist on the `dev` branch and are unrelated to the SummaryBar component

<img width="1164" alt="error-messages" src="https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/83729360/9162d1b3-9471-4adb-af86-eb499b24589c">
